server/jetstream_cluster.go

```diff
@@ -6752,6 +6758,8 @@ RETRY:
 			msgsQ.recycle(&mrecs)
 		case <-notActive.C:
 			if mrecs := msgsQ.pop(); len(mrecs) > 0 {
+				mrec := mrecs[0].(*im)
+				notifyLeaderStopCatchup(mrec, fmt.Errorf("catchup stalled"))
```

Comment:
Make a var error with errors.New?

Reply:
I see we do that even inside functions, which kind of does not make much sense because there will still be a new alloc per invocation. So I think it should be done outside of function (like a global) no?

Reply:
Check last commit where I did extract some of the catchup errors and declared them outside of the funcs.